### PR TITLE
feat(dashboard): claims first implementation

### PR DESCRIPTION
**Pull request by riqwan**
what:

- order claims end to end feature branch

### CODE REVIEW

**Comment by olivermrbl** [comment]
@riqwan, I would be OK with merging this to `develop` incrementally instead of a long-lived feature branch. Wdyt?

**Comment by riqwan** [comment]
@olivermrbl yes, much prefer that. Don't know why this was introduced.

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/claim-create.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const { order: preview } = useOrderPreview(id!)
+
+  const [activeClaimId, setActiveClaimId] = useState()
+
+  const { mutateAsync: createClaim } = useCreateClaim(order.id)
+
+  // TODO: GET /claims/:id is not implemented
+  // const { claim } = useClaim(activeClaimId, undefined, {
+  //   enabled: !!activeClaimId,
+  // })
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/claim-create.tsx:34]: todo: This was introduced yesterday. Maybe just replace it in a follow-up PR

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-inbound-item.tsx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                    )
+                  }}
+                />
+              </div>
+              <IconButton
+                type="button"
+                className="flex-shrink"
+                variant="transparent"
+                onClick={() => {
+                  onUpdate({ reason_id: null }) // TODO BE: we should be able to set to unset reason here
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-inbound-item.tsx:181]: todo: Should be possible as of yesterday

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-inbound-item.tsx [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                  form.setValue(`inbound_items.${index}.reason_id`, "")
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-inbound-item.tsx:182]: nit:
```suggestion
                  form.setValue(`inbound_items.${index}.reason_id`, null)
```

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/schema.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    z.object({
+      item_id: z.string(),
+      quantity: z.number(),
+      reason_id: z.string().optional().nullable(),
+      note: z.string().optional().nullable(),
+    })
+  ),
+  outbound_items: z.array(
+    z.object({
+      item_id: z.string(), // TODO: variant id?
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/schema.ts:14]: comment: Yeah I think we need to be able to pass variants here. We don't have the item ID upfront, as far as I can see. We can tackle that when we add support for outbound items.

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx [hunk 5]
```diff
@@ -0,0 +1,715 @@
...
+
+  /**
+   * HOOKS
+   */
+  const { stock_locations = [] } = useStockLocations({ limit: 999 })
+  const { shipping_options = [] } = useShippingOptions({
+    limit: 999,
+    fields: "*prices,+service_zone.fulfillment_set.location.id",
+    /**
+     * TODO: this should accept filter for location_id
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:81]: question: Do we have a ticket for this?

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx [hunk 6]
```diff
@@ -0,0 +1,715 @@
...
+   */
+
+  const form = useForm<ReturnCreateSchemaType>({
+    defaultValues: () => {
+      const method = preview.shipping_methods.find(
+        (s) => !!s.actions?.find((a) => a.action === "SHIPPING_ADD")
+      )
+
+      return Promise.resolve({
+        inbound_items: previewItems.map((i) => ({
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:161]: question: Will likely need outbound items too when added right?

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx [hunk 7]
```diff
@@ -0,0 +1,715 @@
...
+
+      ;(
+        await Promise.all(
+          items.map(async (_i) => {
+            const item = itemsMap.get(_i.item_id)
+
+            if (!item.variant_id) {
+              return undefined
+            }
+            return await sdk.admin.product.retrieveVariant(
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:320]: suggestion/though: can we use `list`?

**Review by olivermrbl — APPROVED** [review]
Overall LGTM, left a few comments and questions

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/claim-create.tsx (hunk 1 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/orders/order-create-claim/claim-create.tsx:34]: Its already replaced [here](https://github.com/medusajs/medusa/pull/8480/files#diff-9c80b20f3219a9609f9c97f04f2c81a53858a69c5c8ce700b9df73e6381e0d3dR29)

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-inbound-item.tsx (hunk 3 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-inbound-item.tsx:182]: I'll keep this here as is, its changed in my current PR

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx (hunk 5 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:81]: Just created it now

File: packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx (hunk 6 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/orders/order-create-claim/components/claim-create-form/claim-create-form.tsx:161]: yes, haven't gotten to outbound yet.

**Comment by riqwan** [comment]
@olivermrbl Will port over the changes to the current branch i'm working on 🤞🏻

**Pull request merged**